testsuite/tests/typing-objects/Tests.ml

```diff
@@ -222,16 +222,17 @@ and 'a t = 'a t u;;
 Line 2, characters 0-17:
 2 | and 'a t = 'a t u;;
     ^^^^^^^^^^^^^^^^^
-Error: The definition of t contains a cycle:
-       'a t u
+Error: The definition of t contains a cycle
 |}];; (* fails since 4.04 *)
 type 'a u = 'a
 and 'a t = 'a t u;;
 [%%expect{|
 Line 2, characters 0-17:
 2 | and 'a t = 'a t u;;
     ^^^^^^^^^^^^^^^^^
-Error: The type abbreviation t is cyclic
+Error: The type abbreviation t is cyclic:
+         'a t = 'a t u,
+         'a t = 'a t u
```

Comment:
This example and the one below look a bit strange. This comes from the fact that the cycle-detection logic compares the identity of the node of the type-graph, not their structure. The two occurrences of `'a t` here look similar but they are distinct type nodes, so `'a t = 'a t u` is not immediately found to be cyclic. I think that we need to expand `'a t` again to eventually encounter a type whose identity you have already encountered during the traversal.

This is slightly weird for users, but also fairly useful for the people trying to understand the cycle-detection algorithm, and I think not harmful for users either.

Reply:
The new output on this example:

```
Error: The type abbreviation t is cyclic:
         'a t = 'a t u,
         'a t u contains 'a t,
         'a t = 'a t u,
         'a t u = 'a t
```

The same reasoning applies, but it is now easier to follow what is going on.